docs/docs/concepts/Binary_Data.md

```diff
@@ -0,0 +1,150 @@
+
+```js
+#!js api_version=1.0 name=lib
+redis.registerFunction("my_set",
+    (c, key, val) => {
+        return c.call("set", key, val);
+    },
+    {
+        flags: [redis.functionFlags.RAW_ARGUMENTS]
+    }
+);
+```
+
+The above example will allow us to set `key` and `val` even if those are binary data. Run example:
+
+```bash
+127.0.0.1:6379> TFCALL lib.my_set 1 "\xaa" "\xaa"
+"OK"
+127.0.0.1:6379> get "\xaa"
+"\xaa"
+```
+
+Notice that `call` function also except `JS` `ArrayBuffer` arguments.
+
+### Binary Command Results
+
+Getting function arguments as binary data is not enough. We might want to read binary data from Redis key. In order to do this we can use `callRaw` function that will not decode the result as `JS` `String` and instead will return the result as `JS` `ArrayBuffer`. Example:
+
+```js
+#!js api_version=1.0 name=lib
+redis.registerFunction("my_get", 
+    (c, key) => {
+        return c.callRaw("get", key);
+    },
+    {
+        flags: [redis.functionFalgs.RAW_ARGUMENTS]
+    }
+);
+```
+
+The above example will be able to fetch binary data and return it to the user. Run example:
+
+```bash
+27.0.0.1:6379> set "\xaa" "\xaa"
+OK
+127.0.0.1:6379> TFCALL lib.my_get 1 "\xaa"
+"\xaa"
+```
+
+Notice that `JS` `ArrayBuffer` can be returned by RedisGears function, RedisGears will return it to the client as `bulk string`.
+
+### Binary Keys Names On Database Triggers
+
+On [KeySpace triggers](./triggers/KeySpace_Triggers.md), if the key name that triggered the event is binary. The `data.key` field will be NULL. The `data.key_raw` field is always provided as `JS` `ArrayBuffer` and can be used in this case, example:
+
+```js
+#!js api_version=1.0 name=lib
+/* The following is just an example, in general it is discourage to use globals. */
+var n_notifications = 0;
+var last_key = null;
+var last_key_raw = null;
+redis.registerKeySpaceTrigger("consumer", "", function(client, data) {
+    if (data.event == "set") {
+        n_notifications += 1;
+        last_data = data.key;
+        last_key_raw = data.key_raw;
+    }
+});
+
+redis.registerFunction("notifications_stats", async function(){
+    return [
+        n_notifications,
+        last_key,
+        last_key_raw
+    ];
+});
+```
+
+Run example:
+
+```bash
+127.0.0.1:6379> set "\xaa" "\xaa"
+OK
+127.0.0.1:6379> TFCALL lib.notifications_stats 0
+1) (integer) 1
+2) (nil)
+3) "\xaa"
+```
+
+For more information, follow [KeySpace triggers](./triggers/KeySpace_Triggers.md) page.
+
+### Binary Data on Stream Consumers
+
+On [stream triggers](./triggers/Stream_Triggers.md), if the key name is binary. The `data.stream_name` field will be NULL. The `data.stream_name_raw` field is always provided as `JS` `ArrayBuffer` and can be used in this case. In addition, if the content of the steam is binary, it will also appear as `null` under `data.record`. In this case it is possible to use `data.record` (which always exists) and contains the data as `JS` `ArrayBuffer`. Example:
```

Comment:
```suggestion
On [stream triggers](./triggers/Stream_Triggers.md), if the key name is binary. The `data.stream_name` field will be NULL. The `data.stream_name_raw` field is always provided as a `JS` `ArrayBuffer` and can be used in this case. In addition, if the content of the steam is binary, it will also appear as `null` under `data.record`. In this case, it is possible to use `data.record` (which always exists) and contains the data as a `JS` `ArrayBuffer`. Example:
```